src/components/TodoList.tsx

```diff
@@ -0,0 +1,296 @@
+import classNames from 'classnames';
+import React, { useState, useEffect, useRef } from 'react';
+import { Todo } from '../Types';
+import { client } from '../utils/client';
+
+interface Props {
+  todo: Todo[],
+  visibleTodos: Todo[],
+  isLoading: boolean,
+  updatingTodoId: number | null,
+  tempTodo: Todo | null,
+  updateIndividualTodo: (id: number) => Promise<void>,
+  isDoubleClickedName: string,
+  placeHolderText: string,
+  setPlaceHolderText: React.Dispatch<React.SetStateAction<string>>,
+  excludedInputRef: React.RefObject<HTMLInputElement>,
+  isUpdating: boolean,
+  setIsDoubleClickedName: React.Dispatch<React.SetStateAction<string>>,
+  deleteTodo: (id: number) => Promise<void>,
+  setIsUpdating: React.Dispatch<React.SetStateAction<boolean>>,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  isPlusOne: boolean,
+  isThereIssue: boolean,
+  deletedTodoId: number,
+  isEveryThingDelete: boolean,
+  todoStatusChange: boolean,
+  toggleFalseTodosId: number[],
+  isEveryThingTrue: boolean,
+  setIsThereIssue: React.Dispatch<React.SetStateAction<boolean>>,
+  setDeleteErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+}
+
+export const TodoList: React.FC<Props> = ({
```

Comment:
The Props interface contains too many properties. Consider breaking it down into smaller interfaces and use them as needed.

Reply:
Not fixed